src/status_im/utils/utils.cljs

```diff
@@ -76,3 +72,17 @@
 
 (defn clear-interval [id]
   (.clearInterval rn-dependencies/background-timer id))
+
+(defn distinct-by-group
+  "First group a map's values by property <grouper>,
+  then select the optimal within each group as determined
+  by property <optimizing> taking the first after
+   ordering by <ordering>, which defaults to greater-than."
+
+  ([m grouper optimizing ordering]
+   (into {} (map (fn [[dup map-elts]]
+                   (first (sort-by #(optimizing (second %)) ordering map-elts)))
```

Comment:
You could `comp` here for both `grouper` & `optimizing`:
`(comp optimizing second)`
I probably find it a bit more readable, but matter of taste

Reply:
Yes, same with `#(grouper (second %))`

Reply:
Will do.